### ISSUE TITLE
Error when predicting on an image loaded in palleted mode

### ISSUE THREAD

**Issue by hjk789** [ultralytics/ultralytics#1039]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Detection

### Bug

When using an image loaded into memory for prediction, it needs to be read in RGB mode, otherwise it gives the error 
```
File "/home/linuxlite/.local/lib/python3.10/site-packages/ultralytics/yolo/data/dataloaders/stream_loaders.py", line 300, in _single_check
    im = np.asarray(im)[:, :, ::-1]
IndexError: too many indices for array: array is 2-dimensional, but 3 were indexed
```

### Environment

- YOLO: Ultralytics YOLOv8.0.40 🚀 Python-3.10.6 torch-1.13.1+cu117 CPU
- OS: Linux Lite 6.2 on a VM
- Python: 3.10.6

### Minimal Reproducible Example

```
from PIL import Image

img = Image.open("img.png")
YOLO("best.pt").predict(source=img)
```
Which throws

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/linuxlite/.local/lib/python3.10/site-packages/ultralytics/yolo/engine/model.py", line 169, in predict
    return self.predictor.predict_cli(source=source) if is_cli else self.predictor(source=source, stream=stream)
  File "/home/linuxlite/.local/lib/python3.10/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/linuxlite/.local/lib/python3.10/site-packages/ultralytics/yolo/engine/predictor.py", line 113, in __call__
    return list(self.stream_inference(source, model))  # merge list of Result into one
  File "/home/linuxlite/.local/lib/python3.10/site-packages/ultralytics/yolo/engine/predictor.py", line 144, in stream_inference
    self.setup_source(source if source is not None else self.args.source)
  File "/home/linuxlite/.local/lib/python3.10/site-packages/ultralytics/yolo/engine/predictor.py", line 127, in setup_source
    self.dataset = load_inference_source(source=source,
  File "/home/linuxlite/.local/lib/python3.10/site-packages/ultralytics/yolo/data/build.py", line 181, in load_inference_source
    dataset = LoadPilAndNumpy(source, imgsz=imgsz, stride=stride, auto=auto, transforms=transforms)
  File "/home/linuxlite/.local/lib/python3.10/site-packages/ultralytics/yolo/data/dataloaders/stream_loaders.py", line 286, in __init__
    self.im0 = [self._single_check(im) for im in im0]
  File "/home/linuxlite/.local/lib/python3.10/site-packages/ultralytics/yolo/data/dataloaders/stream_loaders.py", line 286, in <listcomp>
    self.im0 = [self._single_check(im) for im in im0]
  File "/home/linuxlite/.local/lib/python3.10/site-packages/ultralytics/yolo/data/dataloaders/stream_loaders.py", line 300, in _single_check
    im = np.asarray(im)[:, :, ::-1]
IndexError: too many indices for array: array is 2-dimensional, but 3 were indexed
```

But works fine with

```
YOLO("best.pt").predict(source=img.convert('RGB'))
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Laughing-q** [comment]
@hjk789 yeah for now we only support RGB mode images, what's the mode of your image? I'd like to reproduce and fix it.

**Comment by hjk789** [comment]
Palleted (P). This error happened when I tried to predict on a PNG image.

**Comment by Laughing-q** [comment]
@hjk789 ok got it! probably I think we should add a `convert("RGB")` here.

**Comment by glenn-jocher** [comment]
@Laughing-q @hjk789 yes thanks for raising this issue. We have this check in YOLOv5 AutoShape here, I think we just need to carry over this code to YOLOv8:
https://github.com/ultralytics/ultralytics/blob/f2a7a29e531ad029255c8ec180ff65de24f42e8d/ultralytics/nn/autoshape.py#L95

**Comment by glenn-jocher** [comment]
@hjk789 this should be resolved in 8.0.42 for release tonight. This code has been changed to:
https://github.com/ultralytics/ultralytics/blob/ae72bcf9b56d52d9821cf41c8b92f8dee678dd03/ultralytics/yolo/data/dataloaders/stream_loaders.py#L299-L302

**Comment by glenn-jocher** [comment]
@hjk789 closing PR as solved in #1078. Please reopen or let us know if this does not resolve your issue, thank you!
